src/Test/Utilities/Portable/Metadata/ILValidation.cs

```diff
@@ -3,16 +3,175 @@
+                var hash = SigningUtilities.CalculateSha1(content);
+
+                var publicKey = CryptoBlobParser.ToRSAParameters(TestResources.General.snKey,
+                    includePrivateParameters: false);
+
+                using (var rsa = RSA.Create())
+                {
+                    rsa.ImportParameters(publicKey);
+                    var reversedSignature = signature.ToArray();
+                    Array.Reverse(reversedSignature);
+                    if (!rsa.VerifyHash(hash, reversedSignature, HashAlgorithmName.SHA1, RSASignaturePadding.Pkcs1))
+                    {
+                        return false;
+                    }
+                }
+
+                return true;
+            }
+        }
+
+        private static MethodInfo s_peheaderSizeMethod;
+        private static int PEHeaderSize(bool is32Bit)
+        {
+            if (s_peheaderSizeMethod == null)
+            {
+                Interlocked.CompareExchange(
+                    ref s_peheaderSizeMethod,
+                    typeof(PEHeader).GetMethod(
+                        "Size",
+                        BindingFlags.Static | BindingFlags.NonPublic),
+                    null);
+            }
+
+            return (int)s_peheaderSizeMethod.Invoke(null, new object[] { is32Bit });
+        }
+
+        private static ConstructorInfo s_blobCtor;
+        private static Blob MakeBlob(byte[] buffer, int offset, int size)
+        {
+            if (s_blobCtor == null)
+            {
+                Interlocked.CompareExchange(
+                    ref s_blobCtor,
+                    typeof(Blob).GetConstructors(
+                        BindingFlags.NonPublic | BindingFlags.Instance).Single(),
+                    null);
+            }
+
+            return (Blob)s_blobCtor.Invoke(new object[] { buffer, offset, size });
+        }
+
+        private static FieldInfo s_bufferField;
+        private static byte[] GetBlobBuffer(Blob blob)
+        {
+            if (s_bufferField == null)
+            {
+                Interlocked.CompareExchange(
+                    ref s_bufferField,
+                    typeof(Blob).GetField(
+                        "Buffer",
+                        BindingFlags.NonPublic | BindingFlags.Instance),
+                    null);
+            }
+
+            return (byte[])s_bufferField.GetValue(blob);
+        }
+
+        private static MethodInfo s_getContentToSignMethod;
+        private static IEnumerable<Blob> GetContentToSign(
+            BlobBuilder peImage,
+            int peHeadersSize,
+            int peHeaderAlignment,
+            Blob strongNameSignatureFixup)
+        {
+            if (s_getContentToSignMethod == null)
+            {
+                Interlocked.CompareExchange(
+                    ref s_getContentToSignMethod,
+                    typeof(PEBuilder).GetMethod(
+                        "GetContentToSign",
```

Comment:
Consider filing an issue to request this API be made public (or a PR to make it public ;-) ).

Reply:
The PI is not gonna be made public. I'll fix the checksum calculation.

Reply:
Instead I added a Debug assert to fire if the checksums are ever equal. If they are, it's likely the bug has been fixed. It may be possible for the assert to fire in a real case where both checksums are the same, but since this assert will be compiled out in release it seemed like an OK risk to take.